queue_/queue_stack_test.py

```
@@ -0,0 +1,115 @@
+        self.assertFalse(queue.empty())
+        self.assertEqual(queue.size(), len(data))
+        self.assertEqual(queue.top(), data[0])
+
+    def test_new_queue_from_generator(self):
+        """
+        Create a QueueStack from a generator.
+        Test that its size equals to the number provided in the generator.
+        """
+        queue = QueueStack(range(10))
+        self.assertFalse(queue.empty())
+        self.assertEqual(queue.size(), 10)
+        self.assertEqual(queue.top(), 0)
+
+    def test_put_element(self):
+        """
+        Put an element in queue_stack.
+        Test that its size is 1.
+        """
+        queue = QueueStack()
+        queue.put(1)
+        self.assertFalse(queue.empty())
+        self.assertEqual(queue.size(), 1)
+        self.assertEqual(queue.top(), 1)
+
+    def test_call_get_of_empty_queue_stack_raised_error(self):
+        """
+        Create an empty QueueStack.
+        Test that call of get function raises Assertion error
+        """
+        queue = QueueStack()
+        self.assertRaises(IndexError, queue.get)
+
+    def test_put_out_of_the_size(self):
+        """
+        Create a QueueStack of certain size. Put more than allowed.
+        Test that the size of QueueStack is unchanged.
+        """
+        queue = QueueStack([1, 2, 3], 3)
+        queue.put(4)
+        self.assertEqual(queue.size(), 3)
+
+    def test_unpassed_size(self):
+        """
+        Create a QueueStack with a not passed size.
+        Test that the size of QueueStack equals the size of data.
+        """
+        queue = QueueStack([1, 2, 3])
+        self.assertEqual(queue.size(), 3)
+
+    def test_put_out_of_the_unpassed_size(self):
+        """
+        Create a QueueStack with a not passed size. Put more than allowed.
+        Test that the size of QueueStack is unchanged.
+        """
+        queue = QueueStack([1, 2, 3])
+        queue.put(2)
+        self.assertEqual(queue.size(), 3)
```

why 3?

---

what do you do when the queue is  full? maybe it's better to raise an error?

---

fixed